pkg/subctl/cmd/join.go

```diff
@@ -112,6 +114,13 @@ func checkArgumentPassed(args []string) error {
 	return nil
 }
 
+func checkBrokerContextPassed(brokerClusterContext string) error {
```

Comment:
This should only be checked if in combination with service discovery.

If you are not doing service discovery this parameter can be not passed.

Reply:
Done. Added the check.